### ISSUE TITLE
Improve block markup for accessibility & SEO

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#9266]
## Feature Description

The Web Stories block markup for things like the list embeds or grid embeds does not contain any link to the story itself. Instead, the URL is just in a `data-story-url` attribute.

![Screenshot 2021-10-05 at 15 58 38](https://user-images.githubusercontent.com/841956/136045507-c1665edd-3fc7-4d81-a9c4-990f08ab6e99.png)

This is not good for accessibility and SEO, and also for when JavaScript is disabled on a page.

Instead, there should be a proper `<a>` tag around the story.

When JS is disabled, this still allows visiting the story by clicking on it.

When JS is enabled, we prevent the click event and open the lightbox as today.

https://github.com/google/web-stories-wp/blob/94aa5519ef7d037ff98f71dec919d07bbd93e9eb/includes/Renderer/Stories/Renderer.php#L521-L523

https://github.com/google/web-stories-wp/blob/94aa5519ef7d037ff98f71dec919d07bbd93e9eb/includes/Renderer/Stories/Renderer.php#L550-L552

https://github.com/google/web-stories-wp/blob/94aa5519ef7d037ff98f71dec919d07bbd93e9eb/includes/Renderer/Stories/Renderer.php#L560-L578

**Comment by swissspidy** [comment]
@csossi FYI for this one, no need for specific testing for the markup. See the PR for instructions

**Comment by csossi** [comment]
Verified in QA (as per instructions in PR 9341)
